centrallix-sysdoc/OSDriver_Authoring.md

```diff
@@ -115,497 +170,1032 @@ License:  Copyright (C) 2001-2011 LightSys Technology Services. See LICENSE.txt
+### Function: Create()
+```c
+int xxxCreate(pObject obj, int mask, pContentType sys_type, char* usr_type, pObjTrxTree* oxt);
+```
+The `Create()` function is used to create a new object, and uses the same parameters and return value as `Open()` (documented in detail above).  This often means adding a new file to the file system to represent the object.  Many drivers do not implement this and recommend that driver end-users create files using a standard text editor or programatically using more general means, such as general structure file generation.  If implemented, this function frequently requires very similar path parsing functionality to `Open()`.
 
-As a side note, within Centrallix, the standard function naming convention is to use xxx_internal_FunctionName for functions that are more or less internal to the module and not a part of any standard interface.
+- 📖 **Note**: For many drivers, the `Create()` function calls the driver's `Open()` function with `O_CREAT`, then calls its `Close()` function, although some drivers may manage this differently.
 
-The Create routine has parameters identical to the Open routine.  It should return 0 on success and -1 on error.
 
-The Delete routine is passed the following parameters:
+### Function: Delete()
+```c
+int clusterDelete(pObject obj, pObjTrxTree* oxt);
+```
+The `Delete()` function is used to delete an object, which often means removing a file from the file system.  The Delete routine is passed the following parameters:
 
 | Param  | Type          | Description
 | ------ | ------------- | ------------
 | obj    | pObject       | The Object structure pointer, used in the same way as in Open and Delete.
-| oxt    | pObjTrxTree*  | The transaction tree pointer.
+| oxt    | pObjTrxTree*  | The transaction tree pointer for the `OBJDRV_C_TRANS` capability.
 
 Delete should return 0 on success and -1 on failure.
 
-For many objectsystem drivers, the Create function simply calls the driver's internal Open() with O_CREAT and then its internal Close, although some drivers could manage Create differently from Open.
 
-### D.  Reading and Writing Object Content.
-Some, but not all, objects will have content.  If the object does or can have content, the driver should handle these functions as is appropriate.  Otherwise, the driver should return a failure code (-1) from these functions.
+### Function: DeleteObj()
+```c
+int xxxDeleteObj(void* inf_v, pObjTrxTree* oxt);
+```
+**No documentation provided.**
+
+
+### Function: Read()
+```c
+int xxxRead(void* inf_v, char* buffer, int max_cnt, int offset, int flags, pObjTrxTree* oxt);
+```
+<!-- TODO: Greg - Above is the signature that the cluster driver uses.  However, this conflicts with the documented parameters below (the `arg` parameter is missing).  Can you please resolve this conflict? -->
```

Comment:
The documentation was incorrect.  The correct parameter order is what you have above.  'arg' was replaced by 'offset', and 'offset' is before the 'flags' argument.

It's probably worth mentioning that the reason that seeking is integrated into the read/write calls is to avoid TOCTOU problems (race conditions).  If an API user wants to just seek only, the API user can do a 0-length read, which is a little non-intuitive, but in most cases it's much better to not have a separate Seek API call.